### ISSUE TITLE
Config options for ignorePath and ignorePattern are being ignored

### ISSUE THREAD

**Issue by fpuga** [stylelint/stylelint#4126]
> Clearly describe the bug

Config options for `ignorePath` and `ignorePattern` are being ignored but `--ignore-path` and `--ignore-pattern` work in cli.

> Which rule, if any, is the bug related to?

None`

> What CSS is needed to reproduce the bug?

None. I attach an example project

[stylelinttest.zip](https://github.com/stylelint/stylelint/files/3314213/stylelinttest.zip)

> What stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
    extends: "stylelint-config-standard",
    rules: {
        indentation: [4, { baseIndentLevel: 1 }],
    },
    ignorePath: ".gitignore",
    ignorePattern: "vendor",
}

```

> Which version of stylelint are you using?

$ stylelint -v
10.1.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

```
# *.css files in .gitignore and 'vendor' are correctly being ignored
stylelint --ignore-path .gitignore --ignore-pattern 'vendor' '**/*.css'

# *.css files in .gitignore and 'vendor' are NOT being ignored
stylelint '**/*.css'

```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

*.css files in .gitignore and 'vendor' are being ignored

> What actually happened (e.g. what warnings or errors did you get)?

*.css files in .gitignore and 'vendor' are NOT being ignored

**Comment by evilebottnawi** [comment]
We don't provide setup ignore in configuration and we don't have docs about this, what is use case to using this option in configuration?

**Comment by Arcanemagus** [comment]
> We don't provide setup ignore in configuration and we don't have docs about this

Actually it's documented as [`ignoreFiles`](https://stylelint.io/user-guide/configuration/#ignorefiles).

@fpuga If you want those files to be ignored _automatically_, you either need use the above option to specify the same globs as in your `.gitignore`, or a [`.stylelintignore` file](https://stylelint.io/user-guide/configuration/#stylelintignore).

**Comment by evilebottnawi** [comment]
@Arcanemagus oh, forgot about this, sorry

**Comment by fpuga** [comment]
Thanks for the comment. I just believe that i can use `ignorePath` and `ignorePattern` in the config file.

My strategy actually is using the `--ignore-path .ignore` from the command line (or as an argument for [pre-commit](https://pre-commit.com/), and share the the `.ignore` file for different formatters like prettier.

I hope that in the future "all" the formatters allow to share a gitignore like file and a common name for the `--ignore-path` option. For example [black is also working in something like that](https://github.com/python/black/issues/475).

Just for others this is some of the config that i'm using.

For pre-commit
```
    - repo: local
      hooks:
          - id: stylelint
            name: stylelint
            entry: npx stylelint
            args: [--fix, --ignore-path, .ignore]
            language: system
            types: [file]
            files: \.(vue|htm|html|css|sss|less|scss|sass|mak|mako|jinja2)$
            verbose: true

          - id: prettier
            name: prettier
            entry: npx prettier
            args: [--write, --ignore-path, .ignore]
            language: system
            types: [file]
            files: \.(js|json|md|vue|yaml|yml)$ # html
            verbose: true
```
In package.json
```
"scripts": {
    "pretty:python": "black . && isort -rc .",
    "pretty:css": "npx stylelint --fix --ignore-path .ignore  '**/*. {vue,htm,html,css,sss,less,scss,sass,mak,jinja2}'",
    "pretty:js": "npx prettier --ignore-path '.ignore' --write '**/*.js'",
    "pretty:others": "npx prettier --ignore-path '.ignore' --write '**/*.{md,yml,yaml,json}'"
```

Thanks for this awesome project, and **feel free to close this issue**.

**Comment by Issayev** [comment]
Does 'ignorePath' option works in js configuration file? 
I've faced with the same problem when --ignore-path works in cli and not with config file.

**Comment by Mouvedia** [comment]
@Issayev https://github.com/stylelint/stylelint/issues/2724#issuecomment-315301853
You could ask for `ignorePath ` to be added to #5142.
